src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const res = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      for (const i in Object.assign(key.extraData)) {
+        StateCopy[i] = key.extraData[i];
+      }
+      res.push({ ...StateCopy });
+    }
+
+    if (key.type === 'removeProperties') {
+      for (const i of Object.assign(key.keysToRemove)) {
```

Comment:
The use of `Object.assign(key.keysToRemove)` is unnecessary here. You can directly iterate over `key.keysToRemove` using `for (const i of key.keysToRemove)`. `Object.assign` is not needed for iteration.